packages/preview2-shim/types/interfaces/wasi-sockets-udp.d.ts

```diff
@@ -29,193 +27,276 @@ export namespace WasiSocketsUdp {
+       * - <https://pubs.opengroup.org/onlinepubs/9699919799/functions/sendmsg.html>
+       * - <https://man7.org/linux/man-pages/man2/send.2.html>
+       * - <https://man7.org/linux/man-pages/man2/sendmmsg.2.html>
+       * - <https://learn.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-send>
+       * - <https://learn.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-sendto>
+       * - <https://learn.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-wsasendmsg>
+       * - <https://man.freebsd.org/cgi/man.cgi?query=send&sektion=2>
+       */
+      /**
+       * Create a `pollable` which will resolve once the stream is ready to send again.
+       * 
+       * Note: this function is here for WASI Preview2 only.
+       * It's planned to be removed when `future` is natively supported in Preview3.
+       */
+    }
+    import type { Pollable } from '../interfaces/wasi-io-poll.js';
+    export { Pollable };
+    import type { Network } from '../interfaces/wasi-sockets-network.js';
+    export { Network };
+    import type { ErrorCode } from '../interfaces/wasi-sockets-network.js';
+    export { ErrorCode };
+    import type { IpSocketAddress } from '../interfaces/wasi-sockets-network.js';
+    export { IpSocketAddress };
+    import type { IpAddressFamily } from '../interfaces/wasi-sockets-network.js';
+    export { IpAddressFamily };
+    /**
+     * A received datagram.
+     */
+    export interface IncomingDatagram {
+      /**
+       * The payload.
+       * 
+       * Theoretical max size: ~64 KiB. In practice, typically less than 1500 bytes.
+       */
+      data: Uint8Array,
+      /**
+       * The source address.
+       * 
+       * This field is guaranteed to match the remote address the stream was initialized with, if any.
+       * 
+       * Equivalent to the `src_addr` out parameter of `recvfrom`.
+       */
+      remoteAddress: IpSocketAddress,
+    }
+    /**
+     * A datagram to be sent out.
+     */
+    export interface OutgoingDatagram {
+      /**
+       * The payload.
+       */
+      data: Uint8Array,
+      /**
+       * The destination address.
+       * 
+       * The requirements on this field depend on how the stream was initialized:
+       * - with a remote address: this field must be None or match the stream's remote address exactly.
+       * - without a remote address: this field is required.
+       * 
+       * If this value is None, the send operation is equivalent to `send` in POSIX. Otherwise it is equivalent to `sendto`.
+       */
+      remoteAddress?: IpSocketAddress,
+    }
+    
+    export class OutgoingDatagramStream {
+      checkSend(): bigint;
+      send(datagrams: OutgoingDatagram[]): bigint;
+      subscribe(): Pollable;
+    }
+    
+    export class IncomingDatagramStream {
+      receive(maxResults: bigint): IncomingDatagram[];
+      subscribe(): Pollable;
+    }
+    
+    export class UdpSocket {
```

Comment:
are me missing the `stream()` defined in the wit file? See https://github.com/bytecodealliance/jco/pull/244/files#r1383077625

Reply:
This PR also updates to the Wasmtime master branch of the wits (via `cd submodules/wasmtime && git fetch origin && git reset --hard origin/main` and then `xtask update preview2` and then `xtask generate wasi-types`).

Reply:
At least on Wasmtime main, this method doesn't exist.